src/util/plotting.jl

```diff
@@ -0,0 +1,64 @@
+using RecipesBase
+
+@recipe f(gp::AbstractGP, x::Array) = gp(x)
+@recipe f(gp::AbstractGP, x::AbstractRange) = gp(x)
+@recipe f(gp::AbstractGP, xmin::Number, xmax::Number) = (gp, range(xmin, xmax, length=1000))
+@recipe function f(gp::FiniteGP; samples=0, sample_seriestype=:line)
+    x = gp.x
+    f = gp.f
+    ms = marginals(gp)
+
+    @series begin
+        μ = mean.(ms)
+        σ = std.(ms)
+        ribbon := σ
+        fillalpha --> 0.3
+        linewidth --> 2
+        x, μ
+    end
+end
+
+"""
+    f(
+        gp::FiniteGP;
+        samples = 1,
+        sample_seriestype = :line,
+        linealpha=0.2,
+        markershape=:circle,
+        markerstrokewidth=0.0,
+        markersize=0.5,
+        markeralpha=0.3,
+        seriescolor="red"
+    )
+"""
+sampleplot
+
+@userplot SamplePlot
+@recipe function f(sp::SamplePlot;
+        samples = 1,
+        sample_seriestype = :line,
+        linealpha=0.2,
+        markershape=:circle,
+        markerstrokewidth=0.0,
+        markersize=0.5,
+        markeralpha=0.3,
+        seriescolor="red"
+    )
+    x = sp.args[1].x
+    f = sp.args[1].f
+    @series begin
+        samples = rand(f(x, 1e-9), samples)
+
+        seriestype --> sample_seriestype
+        linealpha := linealpha
+        markershape --> markershape
+        markerstrokewidth --> markerstrokewidth
+        markersize --> markersize
+        markeralpha --> markeralpha
+        seriescolor --> seriescolor
+        label := ""
```

Comment:
As mentioned above, defaults should be provided in the series definition:
```suggestion
        seriestype --> :line
        linealpha --> 0.2
        markershape --> :circle
        markerstrokewidth --> 0.0
        markersize --> 0.5
        markeralpha --> 0.3
        seriescolor --> :red
        label --> ""
```

Reply:
Sorry, I don't seem to understand. Don't we intend to the give the user the flexibility to change the defaults? If we were to set the defaults in the series definition,  how would the user modify them when required?

Reply:
That's all handled by the macros in RecipesBase. If you specify plotting options by `setting --> defaultvalue` then a user can specify a different setting by using the keyword argument `setting = someothervalue`. However, if you write `setting := value` then users can't specify a different value and it's fixed to `value`.